### PR TITLE
[Will not merge to main] Labels rework on the resource `google_dataproc_cluster`

### PR DESCRIPTION
**Pull request by zli82016**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dataproc : changed `labels` field to only manage labels configured on the resource `google_dataproc_cluster`
```

```release-note:enhancement
dataproc : added `effective_labels` field to `google_dataproc_cluster` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_dataproc_cluster_test.go.erb [hunk 1]
```diff
@@ -690,11 +690,69 @@ func TestAccDataprocCluster_withLabels(t *testing.T) {
...
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy: testAccCheckDataprocClusterDestroy(t),
+		Steps: []resource.TestStep{
+			{
+				Config:                   testAccDataprocCluster_withoutLabels(rnd),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDataprocClusterExists(t, "google_dataproc_cluster.with_labels", &cluster),
+
+					resource.TestCheckNoResourceAttr("google_dataproc_cluster.with_labels", "labels.%"),
+					// We only provide one, but GCP adds three and goog-dataproc-autozone is added internally, so expect 4.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_dataproc_cluster_test.go.erb:730]:
```suggestion
					// We don't provide any but GCP adds three and goog-dataproc-autozone is added internally, so expect 4.
```

File: mmv1/third_party/terraform/tests/resource_dataproc_cluster_test.go.erb [hunk 2]
```diff
@@ -690,11 +690,69 @@ func TestAccDataprocCluster_withLabels(t *testing.T) {
...
+					// We only provide two, so expect 2.
+					resource.TestCheckResourceAttr("google_dataproc_cluster.with_labels", "labels.%", "1"),
+					resource.TestCheckResourceAttr("google_dataproc_cluster.with_labels", "labels.key2", "value2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataprocCluster_withLabelsUpdate(t *testing.T) {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_dataproc_cluster_test.go.erb:712]: Can we just supersede `TestAccDataprocCluster_withLabels` with this test case by testing it between `testAccDataprocCluster_withLabels` and `testAccDataprocCluster_withoutLabels`? It's not clear what we're getting out of having both to me.

File: mmv1/third_party/terraform/tests/resource_dataproc_cluster_upgrade_test.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+				ExternalProviders: oldVersion,
+			},
+			{
+				Config:                   testAccDataprocCluster_withoutLabels(rnd),
+				ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDataprocClusterExists(t, "google_dataproc_cluster.with_labels", &cluster),
+
+					resource.TestCheckNoResourceAttr("google_dataproc_cluster.with_labels", "labels.%"),
+					// We only provide one, but GCP adds three and goog-dataproc-autozone is added internally, so expect 4.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_dataproc_cluster_upgrade_test.go:40]: We provide none?

**Review by rileykarson — COMMENTED** [review]
LGTM! I'm gonna avoid clicking the approve button until we change the branch target, just for an extra level of safety haha.

File: mmv1/third_party/terraform/tests/resource_dataproc_cluster_upgrade_test.go (hunk 3 above)
Comment (zli82016) [reply mmv1/third_party/terraform/tests/resource_dataproc_cluster_upgrade_test.go:40]: Right. Thanks for catching it.

File: mmv1/third_party/terraform/tests/resource_dataproc_cluster_test.go.erb (hunk 2 above)
Comment (zli82016) [reply mmv1/third_party/terraform/tests/resource_dataproc_cluster_test.go.erb]: Done.

File: mmv1/third_party/terraform/tests/resource_dataproc_cluster_test.go.erb (hunk 1 above)
Comment (zli82016) [reply mmv1/third_party/terraform/tests/resource_dataproc_cluster_test.go.erb]: Done.

**Comment by zli82016** [comment]
Will include it in a new PR.

**Pull request closed without merging**